src/ComponentBase.ts

```diff
@@ -165,11 +165,11 @@ abstract class ComponentBase<P extends React.Props<any>, S extends Object> exten
         this._isMounted = false;
     }
 
-    componentWillUpdate(nextProps: P, nextState: S): void {
+    componentWillUpdate(nextProps: Readonly<P>, nextState: S, nextContext: any): void {
```

Comment:
should nextState also be readonly as per typings? (for all)

Reply:
fixed